pkg/migrations/sql/idempotent/base.sql

```diff
@@ -1724,71 +1756,126 @@ BEGIN
+    END IF;
+
     FOR r IN
         SELECT *
-        FROM SCHEMA_CATALOG.get_metrics_that_need_drop_chunk()
+        FROM unnest(remaining_metrics)
     LOOP
+        PERFORM set_config('application_name', format('promscale maintenance: data retention: metric %s: wait for lock', r.metric_name), false);
         PERFORM SCHEMA_CATALOG.lock_metric_for_maintenance(r.id);
-        CALL SCHEMA_CATALOG.drop_metric_chunks(r.metric_name, NOW() - SCHEMA_CATALOG.get_metric_retention_period(r.metric_name));
+        CALL SCHEMA_CATALOG.drop_metric_chunks(r.metric_name, NOW() - SCHEMA_CATALOG.get_metric_retention_period(r.metric_name), log_verbose=>log_verbose);
         PERFORM SCHEMA_CATALOG.unlock_metric_for_maintenance(r.id);
 
         COMMIT;
     END LOOP;
 END;
 $$ LANGUAGE PLPGSQL;
-COMMENT ON PROCEDURE SCHEMA_CATALOG.execute_data_retention_policy()
+COMMENT ON PROCEDURE SCHEMA_CATALOG.execute_data_retention_policy(boolean)
 IS 'drops old data according to the data retention policy. This procedure should be run regularly in a cron job';
-GRANT EXECUTE ON PROCEDURE SCHEMA_CATALOG.execute_data_retention_policy() TO prom_maintenance;
+GRANT EXECUTE ON PROCEDURE SCHEMA_CATALOG.execute_data_retention_policy(boolean) TO prom_maintenance;
 
 --public procedure to be called by cron
 --right now just does data retention but name is generic so that
 --we can add stuff later without needing people to change their cron scripts
 --should be the last thing run in a session so that all session locks
 --are guaranteed released on error.
-CREATE OR REPLACE PROCEDURE SCHEMA_PROM.execute_maintenance()
+CREATE OR REPLACE PROCEDURE SCHEMA_PROM.execute_maintenance(log_verbose boolean = false)
 AS $$
+DECLARE
+   startT TIMESTAMPTZ;
 BEGIN
-    CALL SCHEMA_CATALOG.execute_data_retention_policy();
+    startT := clock_timestamp();
+    IF log_verbose THEN
+        RAISE LOG 'promscale maintenance: data retention: starting';
+    END IF;
+    PERFORM set_config('application_name', format('promscale maintenance: data retention'), false);
+    CALL SCHEMA_CATALOG.execute_data_retention_policy(log_verbose=>log_verbose);
     IF SCHEMA_CATALOG.get_timescale_major_version() >= 2 THEN
-        CALL SCHEMA_CATALOG.execute_compression_policy();
+        IF log_verbose THEN
+            RAISE LOG 'promscale maintenance: compression: starting';
+        END IF;
+        PERFORM set_config('application_name', format('promscale maintenance: compression'), false);
+        CALL SCHEMA_CATALOG.execute_compression_policy(log_verbose=>log_verbose);
+    END IF;
+    IF log_verbose THEN
+        RAISE LOG 'promscale maintenance: finished in %', clock_timestamp()-startT;
+    END IF;
+    IF clock_timestamp()-startT > INTERVAL '12 hours' THEN
+        RAISE WARNING 'promscale maintenance jobs are taking too long (one run took %)', clock_timestamp()-startT
+              USING HINT = 'Please consider increasing the number of maintenance jobs using config_maintenance_jobs()';
```

Comment:
Should we also add background workers?

```suggestion
              USING HINT = 'Please consider increasing the number of maintenance jobs using config_maintenance_jobs() or timescaledb.background_workers';
```

Reply:
I don't think so as the default background workers is so much higher and its less likely people hit this